Console/retro/Console.cc

```diff
@@ -292,35 +299,61 @@ void Console::ProcessEscSequence(char c)
             currentAttr.setUnderline(true);
             break;
         default:
-            isProcessingEscSequence=false;
+            sequenceState=State::noSequence;   // Unrecognized sequence
         }
         break;
-    case 2:
+    case State::waitingForM:
         if(c=='m')
-            isProcessingEscSequence=false;
-        else if(c==';')
-            sequenceStep=1;
+            sequenceState=State::noSequence;   // Normal end of sequence
+        else
+            sequenceState=State::noSequence;   // Unrecognized sequence (but we end it anyway!)
+        break;
+    case State::waitingForOSCStart:
+        if(c=='0')
+            sequenceState=State::waitingForSemicolon;
         else
-            isProcessingEscSequence=false;
+            sequenceState=State::noSequence;   // Normal end of sequence
+        break;
+    case State::waitingForSemicolon:
+        if(c==';')
+        {
+            sequenceState=State::inWindowName;
+            windowName="";
+        }
+        else
+            sequenceState=State::noSequence;   // Normal end of sequence
+        break;
+    case State::inWindowName:
+        if(c==BEL)
+        {
+            setWindowName(std::move(windowName));
+            sequenceState=State::noSequence;   // Normal end of sequence
+        }
+        else
+        {
+            if(windowName.size() < MAX_LEN)
+                windowName+=c;
+        }
         break;
     default:
-        sequenceStep=0;
+        sequenceState=State::noSequence;
+        break;
     }
+    return true;
 }
 
 void Console::PutCharNoUpdate(char c)
 {
-    if(isProcessingEscSequence)
-    {
-        ProcessEscSequence(c);
+    if(ProcessEscSequence(c))
         return;
-    }
+
     InvalidateCursor();
     switch(c)
     {
     case '\033':    // Begin of an ANSI escape sequence
-        isProcessingEscSequence=true;
-        sequenceStep=0;
+        //isProcessingEscSequence=true;
```

Comment:
I know there is some commented-out code that I left somewhere else in this source file. I really should remove that, but these lines should go, too :-)